modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -425,10 +430,16 @@ class PublishDir {
             
             if( !sameRealPath && shouldOverwrite(source, destination) ) {
                 log.warn "Overwriting file at ${destination.toUriString()}"
-                FileHelper.deletePath(destination)
-                processFileImpl(source, destination)
+
+                if (destination.getFileSystem().provider().getScheme().equals("latch")) {
+                    processFileImpl(source, destination, true)
+                } else {
+                    FileHelper.deletePath(destination)
+                    processFileImpl(source, destination)
+                }
```

Comment:
```suggestion
                if (!destination.getFileSystem().provider().getScheme().equals("latch")) {
                    FileHelper.deletePath(destination)
                }
                
                processFileImpl(source, destination)
```

Reply:
the processFileImpl call is also different in both cases which is why i did it this way